presto-orc/src/main/java/com/facebook/presto/orc/writer/FlatMapColumnWriter.java

```diff
@@ -0,0 +1,289 @@
+            Type type,
+            List<OrcType> orcTypes,
+            ColumnWriterFactory valueWriterfactory)
+    {
+        checkArgument(column >= 0, "column is negative");
+        checkArgument(sequence >= 0, "sequence is negative");
+
+        this.column = column;
+        this.sequence = sequence;
+        this.keyColumnId = orcTypes.get(column).getFieldTypeIndex(0);
+        this.valueColumnId = orcTypes.get(column).getFieldTypeIndex(1);
+
+        this.type = (MapType) type;
+        this.keyType = type.getTypeParameters().get(0);
+        this.valueType = type.getTypeParameters().get(1);
+
+        this.keyOrcType = orcTypes.get(keyColumnId);
+
+        this.compressed = columnWriterOptions.getCompressionKind() != NONE;
+
+        this.columnEncoding = new ColumnEncoding(orcEncoding == DWRF ? DWRF_MAP_FLAT : DIRECT_V2, 0);
+        this.keyColumnEncoding = new ColumnEncoding(orcEncoding == DWRF ? DIRECT : DIRECT_V2, 0);
+        this.valueColumnEncoding = new ColumnEncoding(orcEncoding == DWRF ? DIRECT : DIRECT_V2, 0);
+
+        this.dwrfEncryptor = dwrfEncryptor;
+        this.metadataWriter = new CompressedMetadataWriter(metadataWriter, columnWriterOptions, dwrfEncryptor);
+        this.inputMetadataWriter = metadataWriter;
+
+        this.valueWriterFactory = valueWriterfactory;
+        this.stringStatisticsLimitInBytes = toIntExact(columnWriterOptions.getStringStatisticsLimit().toBytes());
+
+
+        this.lengthStream = createLengthOutputStream(columnWriterOptions, dwrfEncryptor, orcEncoding);
+        this.presentStream = new PresentOutputStream(columnWriterOptions, dwrfEncryptor);
+        this.keyNodes = new ArrayList<>();
+
+        switch (keyOrcType.getOrcTypeKind()) {
+            case CHAR:
+                checkArgument(orcEncoding != DWRF, "DWRF does not support %s type", type);
+                // fall through
+            case VARCHAR:
+            case STRING:
+                this.blockWriter = new FlatMapStringKeyBlockWriter(this.valueColumnId, columnWriterOptions, dwrfEncryptor, inputMetadataWriter, type, this.keyOrcType, this.valueWriterFactory, this.keyNodes);
+                this.statisticsBuilderSupplier = this::newStringStatisticsBuilder;
+                break;
+            case BYTE:
+            case SHORT:
+            case INT:
+            case LONG:
+                this.blockWriter = new FlatMapNumericKeyBlockWriter(this.valueColumnId, columnWriterOptions, dwrfEncryptor, inputMetadataWriter, type, this.keyOrcType, this.valueWriterFactory, this.keyNodes);
+                this.statisticsBuilderSupplier = IntegerStatisticsBuilder::new;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported type for FlatMap Keys : " + type);
+        }
+        this.statisticsBuilder = statisticsBuilderSupplier.get();
+    }
+
+    @Override
+    public List<ColumnWriter> getNestedColumnWriters()
+    {
+        return ImmutableList.of();
```

Comment:
Why is this returning empty map ?

Reply:
We do not have any nested ColumnWriters.
The key type has no column Writers and the value type creates Column Writers for sequences only after looking at the data.